ganga/GangaCore/Core/GangaRepository/DStreamer.py

```diff
@@ -0,0 +1,529 @@
+                        component_obj, attr, temp_val
+                    )
+                    if local_error:
+                        errors.append(local_error)
+                elif (
+                    isinstance(part_attr[attr],
+                               dict) and "category" in part_attr[attr]
+                ):
+                    component_obj, local_error = JsonLoader.load_component_object(
+                        component_obj, attr, part_attr[attr]
+                    )
+                else:
+                    try:
+                        component_obj.setSchemaAttribute(attr, part_attr[attr])
+                    except Exception as e:
+                        errors.append(e)
+                        raise GangaException(
+                            "ERROR in loading Json, failed to set attribute %s for class %s"
+                            % (attr, type(component_obj))
+                        )
+
+        # Assigning the component object to the master object
+        parent_obj.setSchemaAttribute(name, component_obj)
+        # I do not think we should return error, as it is already raised before
+        return parent_obj, errors
+
+    @staticmethod
+    def load_list_object(part_attr):
+        """Loading component objects that will be attached to a list object
+        """
+        errors = []
+        try:
+            component_obj = allPlugins.find(
+                part_attr["category"], part_attr["type"]
+            ).getNew()
+        except PluginManagerError as e:  # TODO: Maybe move this to the logger
+            print(e)
+            errors.append(e)
+            component_obj = EmptyGangaObject()
+
+        # Assigning the component object its attributes
+        for attr, item in component_obj._schema.allItems():
+            if attr in part_attr:
+                try:
+                    component_obj.setSchemaAttribute(attr, part_attr[attr])
+                except Exception as e:
+                    errors.append(e)
+                    raise GangaException(
+                        "ERROR in loading Json, failed to set attribute %s for class %s"
+                        % (attr, type(component_obj))
+                    )
+
+        return component_obj, errors
+
+    @staticmethod
+    def load_simple_object(parent_obj, name, value):
+        """Attaching a simple attribute to the ganga object
+        """
+        errors = []
+        try:
+            parent_obj.setSchemaAttribute(name, value)
+        except Exception as e:
+            errors.append(e)
+            raise GangaException(
+                "ERROR in loading Json, failed to set attribute %s for class %s"
+                % (name, type(parent_obj))
+            )
+        return parent_obj, errors
+
+
+# TODO: Add more functions here
+class XmlToDatabaseConverter:
+    """This will ensure full backwards compatibilty. Functions for creating LocalJson repo from LocalXML and vice versa.
```

Comment:
I do not think that we need to be able to change json representation in XML. In general we only look for compatibility in one direction.